### DISCUSSION TITLE
Animations using build123d

### DISCUSSION THREAD

**Discussion by Miguel-N** [gumyr/build123d#730] [Q&A]
### Is there a way to make good animations using build123d?

I'm interested in animating mechanisms.

So far, i have tried to just move a box using the move commands and view the animation in the ocp-vscode viewer, but i think this is not the right approach, as the movement doesn't feel fluid.
<img src="https://i.ytimg.com/vi/DrpvQ4_XJro/oar2.jpg" height="200">

[Animation using build123d and OCP CAD Viewer](https://youtube.com/shorts/DrpvQ4_XJro?feature=share)

Previously, I tried a similar approach to animate a fourbar mechanism using FreeCAD to import step files and updating the positions of the parts based on the results of an optimized complex algebra quick solver i made, and the results seem fine, but when the complexity of the step files increases, the visualization is very buggy. From what i could gather, this is due to the backend FreeCAD uses to render the models.
<img src="https://i.ytimg.com/vi/liJQEVI6XaI/oar2.jpg" height="200">

[Fourbar mechanism animation using FreeCAD](https://youtube.com/shorts/liJQEVI6XaI?si=7vy3f8TsQI1csEEg)

Is there a viewer that works with build123d that would give me fluid animations?
Or is there an animation file format that i can use to export an animation using build123d?

Any help would be appreciated.

**Comment by bernhard-42** [comment]
You can try https://github.com/bernhard-42/bd_animation, it uses the threejs animation system for fluid animations. I added some gifs in the Readme to show some results
It needs to be installed from github, there is no pypi lib yet.
